File: week07_0915/전윤지/Bj15829.py
import sys

input = sys.stdin.readline
MOD = 1234567891
R = 31

L = int(input().strip())          
s = input().strip()               

h = 0                             
r_pow = 1                         

for ch in s:                      
    val = ord(ch) - 96            
    h = (h + val * r_pow) % MOD   # h += a_i * R^i  (모듈러 적용)
    r_pow = (r_pow * R) % MOD     # 다음 i = R^(i+1)

print(h)

File: week07_0915/전윤지/Bj15829.java
package week07_0915.전윤지;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Bj15829 {
    public static void main(String[] args) throws IOException {
        final int R = 31;
        final long MOD = 1234567891L;

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int L = Integer.parseInt(br.readLine());
        String s = br.readLine();

        long hash = 0;
        long rPow = 1; // R^0 = 1

        for (int i = 0; i < L; i++) {
            int val = s.charAt(i) - 'a' + 1; // a=1, b=2, ..., z=26
            hash = (hash + val * rPow) % MOD;
            rPow = (rPow * R) % MOD;
        }

        System.out.println(hash);
    }

}
